protocols/autonatv2/src/server/handler/dial_request.rs

```diff
@@ -0,0 +1,355 @@
+use std::{
+    io,
+    sync::Arc,
+    task::{Context, Poll},
+    time::Duration,
+};
+
+use either::Either;
+use futures::{
+    channel::{mpsc, oneshot},
+    AsyncRead, AsyncWrite, SinkExt, StreamExt,
+};
+use futures_bounded::FuturesSet;
+use libp2p_core::{
+    upgrade::{DeniedUpgrade, ReadyUpgrade},
+    Multiaddr,
+};
+use libp2p_identity::PeerId;
+use libp2p_swarm::{
+    handler::{ConnectionEvent, FullyNegotiatedInbound, ListenUpgradeError},
+    ConnectionHandler, ConnectionHandlerEvent, StreamProtocol, SubstreamProtocol,
+};
+use rand_core::RngCore;
+
+use crate::{
+    generated::structs::{mod_DialResponse::ResponseStatus, DialStatus},
+    request_response::{DialDataRequest, DialRequest, DialResponse, Request, Response},
+    Nonce,
+};
+use crate::{request_response::Coder, REQUEST_PROTOCOL_NAME};
+
+#[derive(Clone, Debug)]
+pub struct StatusUpdate {
```

Comment:
I'd like the type that we return from `NetworkBehaviour::ToSwarm` to be defined within `behaviour.rs`. Reason being that whatever is public API should be defined as high up as possible.

Reply:
Changed